src/com/facebook/buck/parser/UnexpectedFlavorException.java

```diff
@@ -105,22 +107,32 @@ public static UnexpectedFlavorException createWithSuggestions(
   }
 
   private static ImmutableSet<Flavor> getInvalidFlavors(Flavored flavored, BuildTarget target) {
-    ImmutableSet.Builder<Flavor> builder = ImmutableSet.builder();
-    for (Flavor flavor : target.getFlavors()) {
-      if (!flavored.hasFlavors(ImmutableSet.of(flavor))) {
-        builder.add(flavor);
-      }
-    }
-    return builder.build();
+    return target
+        .getFlavors()
+        .stream()
+        .filter(flavor -> !flavored.hasFlavors(ImmutableSet.of(flavor)))
+        .collect(MoreCollectors.toImmutableSet());
+  }
+
+  private static ImmutableSet<Flavor> getValidFlavors(Flavored flavored, BuildTarget target) {
+    return target
+        .getFlavors()
+        .stream()
+        .filter(flavor -> flavored.hasFlavors(ImmutableSet.of(flavor)))
+        .collect(MoreCollectors.toImmutableSet());
   }
 
   private static String createDefaultMessage(
-      BuildTarget target, ImmutableSet<Flavor> invalidFlavors) {
-    ImmutableSet<String> invalidFlavorsStr =
-        invalidFlavors.stream().map(Flavor::toString).collect(MoreCollectors.toImmutableSet());
+      BuildTarget target, ImmutableSet<Flavor> invalidFlavors, ImmutableSet<Flavor> validFlavors) {
+    String invalidFlavorsStr =
+        invalidFlavors.stream().map(Flavor::toString).collect(Collectors.joining("\n"));
```

Comment:
please use a cross platform way of generating new lines `System.lineSeparator()`